### ISSUE TITLE
Auto-completion list not focused correctly

### ISSUE THREAD

**Issue by EuropeanDeveloper** [notepad-plus-plus/notepad-plus-plus#11415]
### Description of the Issue

The focus in auto-completion list should be on the first matching item but it is on the first item starting with the same character case as input.

### Steps to Reproduce the Issue

1. Open empty file.
2. Set language to `JavaScript`.
3. Type `r`.

### Expected Behavior

The focus should be on the first matching item (`Radio`).

### Actual Behavior

The focus is on the first item starting with the same character case as input (`random`).
 
![screenshot](https://user-images.githubusercontent.com/52759436/159172139-18ce4f57-203a-49c4-b290-836cf4b3ccdb.png)

### Debug Information

Notepad++ v8.3.3   (64-bit)
Build time : Mar 13 2022 - 17:20:02
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19042.1526
Current ANSI codepage : 1250
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll XMLTools.dll

**Comment by ArkadiuszMichalski** [comment]
JavaScript is case sensitive so this behavior looks like intentional. Type `R` and you get `Radio`. Although, for example, typing `ra` returns` Radio`, so it's not so clear cut.

**Comment by EuropeanDeveloper** [comment]
According to [JavaScript auto-completion list](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/e87e94a6cef54f61605dd205051e0ed9e1725b80/PowerEditor/installer/APIs/javascript.xml), it is case <ins>in</ins>sensitive. It has no `<Environment>` tag. See [user manual](https://npp-user-manual.org/docs/auto-completion/#auto-completion-file-format).

**Comment by ArkadiuszMichalski** [comment]
Right, I thought it was fixed earlier as discussed https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4998. JS list should have:
`<Environment ignoreCase="no" startFunc="(" stopFunc=")" paramSeparator="," terminal=";" />`
and works as expected.

@donho Looks like you were supposed to take a look at it https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4998#issuecomment-696458439, but somehow missed it. Some languages are case sensitive but still don't have the `ignoreCase` attribute set.
